pkg/ioutils/ioutils_test.go

```diff
@@ -41,3 +42,33 @@ func (s *MySuite) TestIoutils(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(status, Equals, true)
 }
+
+func TestParseDurationTime(t *testing.T) {
+
+	testCases := []struct {
+		timeValue string
+		expected  time.Duration
+		err       string
+	}{
+		// Test 1: empty string as input
+		{"", 0, "time: invalid duration "},
+		// Test 2: Input string contains 4 day, 10 hour and 3 minutes
+		{"4d10h3m", 900180000000000, ""},
+		// Test 3: Input string contains 10 day and  3 hours
+		{"10d3h", 2170800000000000, ""},
+		// Test 4: Input string contains minutes and days
+		{"3m7d", 1512180000000000, ""},
+		// Test 5: Input string contains unknown unit
+		{"4a3d", 0, "time: unknown unit a in duration 4a3d"},
```

Comment:
I am not sure whether we should support fractional values like `1.5d`. I see `time.ParseDuration` already supports it .
 
```complex, _ := time.ParseDuration("1.5h10m10s")```

Reply:
looks like our ParseDurationTime gives inconsistent results for such fractional values. It gives me `90h0m0s` for `1.5d`

Reply:
@Praveenrajmani Modified. PTAL